solr/core/src/java/org/apache/solr/cloud/api/collections/CreateCollectionCmd.java

```diff
@@ -321,6 +319,8 @@ public void call(ClusterState clusterState, ZkNodeProps message, @SuppressWarnin
         }
       }
       if(isPRS) {
+        byte[] data = Utils.toJSON(Collections.singletonMap(collectionName, clusterState.getCollection(collectionName)));
+        ocmh.zkStateReader.getZkClient().setData(collectionPath, data, true);
```

Comment:
Can we consider using `ocmh.overseer.offerStateUpdate` here instead of direct writes? There are race conditions elsewhere even within the same OS node that some steps are trying to do direct writes while other steps uses the overseer queue. 

I think it's probably safer to at least avoid all direct writes of state.json

Reply:
agree with @patsonluk. Also observed same in my other PR for testListener!